### PR TITLE
[NOT READY] Integrate converage reporting

### PR DESCRIPTION
**Pull request by TheMarex**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -133,6 +134,10 @@ if(CMAKE_BUILD_TYPE MATCHES Release)
   endif()
 endif()
 
+if (COVERAGE)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftest-coverage")
```
Comment (daniel-j-h) [inline CMakeLists.txt]: Do we need `-g -O0` or at least `-fno-inline` in order for this to work properly?

Comment (daniel-j-h) [inline CMakeLists.txt]: Ah, and `-fprofile-arcs` for storing arcs between basic blocks in the data flow graph.

**Comment by daniel-j-h** [comment]
See https://github.com/Project-OSRM/osrm-backend/issues/1673, and close ticket if you merge this in.

File: CMakeLists.txt (hunk 1 above)
Comment (TheMarex) [inline CMakeLists.txt]: I'm going to add a check for `CMAKE_BUILD_TYPE == Debug`. Because that is the only sensible combination.

**Comment by TheMarex** [comment]
:tada: :

**Pull request merged**